src/react-components/src/components/02-molecules/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,82 @@
+// MF-46 Search Bar with Search By Parameter
+
+import * as React from "react";
+import bem from "../../../utils/bem";
+import FormDropdown from "../../01-atoms/Forms/FormDropdown";
+import TextField from "../../01-atoms/Forms/TextField";
+import { Button } from "../../..";
+
+export interface SearchBarProps {
+  searchBarId: string;
+  dropdownId?: string;
+  dropdownOptions?: string[];
+  selectedField?: string;
+  searchValue?: string;
+  placeHolder?: string;
+  searchButtonAttributes?: {};
+  placeholderText?: string;
+  textFieldAttributes?: {};
+  hasError?: boolean;
+  errorMessage?: string;
+  selectBlurHandler?: (event: React.MouseEvent) => void;
+  selectChangeHandler?: (event: React.MouseEvent) => void;
+  searchSubmitHandler: (event: React.MouseEvent) => void;
+  searchChangeHandler?: (event: React.FormEvent) => void;
+}
+
+export default function SearchBar(props: SearchBarProps) {
+
+  const { searchBarId, dropdownId, dropdownOptions, selectedField, placeholderText, hasError, errorMessage,
+    selectBlurHandler, searchSubmitHandler, selectChangeHandler, searchChangeHandler } = props;
+
+  if (dropdownOptions) {
+    if (!(dropdownId && selectBlurHandler && selectChangeHandler)) {
+      throw new Error("If dropdownOptions are passed, dropdownId, selectChangeHandler, and selectBlurHandler must also be passed");
+    }
+  }
+  let modifiers = [];
+  if (hasError) {
+    if (!errorMessage) {
+      throw new Error("If there is an error, there must also be an error message");
+    }
+    modifiers.push("error");
+  }
+
+  let searchbar__base_class = "search-bar";
+
+  let textfieldProps = {
+    labelId: searchBarId,
+    onChange: searchChangeHandler,
+    isRequired: true,
+    blockName: searchbar__base_class,
+    placeholderText: placeholderText,
+    modifiers: modifiers,
+  };
+
+  let buttonProps = {
+    id: searchBarId,
+    callback: searchSubmitHandler,
+    blockName: searchbar__base_class,
+    content: "Search",
+    type: "filled",
```

Comment:
Reading this, I thought that `type` was the HTML attribute for the button, but it's not. Maybe reconsider naming this slightly different to not confuse with the `type` attribute for a `button` (i.e. the type in `<button type="submit" />`). This is NOT for this PR but just a comment.

Reply:
I made a note here:  https://github.com/NYPL/nypl-design-system/projects/2#card-30647396

Reply:
We could potentially use `modifier-type`, it's what we're calling those for the CSS